### PR TITLE
feat: only delete capi cluster if owned

### PR DESCRIPTION
**Pull request by richardcase**
This change will ensure that the CAPI  **Cluster** custom resource is not deleted when deleting the provisioningv2 **Cluster** custom resource with the same name.

## Problem
When using Rancher in this situation:

-  the feature flag to disable the embedded CAPI controllers (`embedded-cluster-api=false`) is set
-  CAPI ins installed  into the Rancher Manager cluster 
- provision a cluster using CAPI 
- import this cluster into CAPI by creating the required `provisioning.cattle.io/v1/Cluster` manually, waiting for the registration token and then getting the import manifests into the CAPI created cluster

If you then delete the `provisioning.cattle.io/v1/Cluster` from the Rancher Manager cluster it will delete any CAPI Cluster with the same name, even though the CAPI cluster is not owned by Rancher. This then causes CAPI to delete the cluster.
 
## Solution
The solution is to only delete the CAPI cluster if it was created by Rancher in the first place. This is determined through the `ownerReferences` on the CAPI cluster.
 
## Testing
The repo steps are valid from the issue.

### Manual Testing

I did these steps to test:

- Created a 2.7.0 Rancher cluster with the feature flag `embedded-cluster-api=false`
- Installed the CAPI CRDs (not the controllers)
- Created an instance of `Cluster.cluster.x-k8s.io` named  'richtest1`
- Created an instance of `Cluster.provisioning.cattle.io` named `richtest1`
- Deleted the instance of `Cluster.provisioning.cattle.io`
- Observed that the `Cluster.cluster.x-k8s.io`  instance is NOT deleted

### Automated Testing
Unit tests have been added to cover this specific functionality

### CODE REVIEW

**Review by Oats87 — DISMISSED** [review]
LGTM; for posterity purposes, would you mind opening a backport PR to `release/v2.6` once this is merged?

**Review by jakefhyde — APPROVED** [review]

**Comment by richardcase** [comment]
> LGTM; for posterity purposes, would you mind opening a backport PR to `release/v2.6` once this is merged?

@Oats87 - yes no problem at all, will so that now.

File: pkg/controllers/provisioningv2/cluster/remove.go [hunk 1]
```diff
@@ -130,3 +133,17 @@ func (h *handler) doClusterRemove(cluster *v1.Cluster) func() (string, error) {
 		return "", h.kubeconfigManager.DeleteUser(cluster.Namespace, cluster.Name)
 	}
 }
+
+func capiClusterIsRancherOwned(cluster *v1.Cluster, capiCluster *capi.Cluster) bool {
```
Comment (alexander-demicev) [inline pkg/controllers/provisioningv2/cluster/remove.go:137]: is there a possibility that any argument will be nil?

Comment (richardcase) [reply pkg/controllers/provisioningv2/cluster/remove.go:137]: No, it would've failed before it got here.

**Comment by richardcase** [comment]
Fixes: #39924

**Comment by richardcase** [comment]
@Oats87 @jakefhyde - is there a way for me to rerun the drone CI?

**Comment by richardcase** [comment]
Thanks @jakefhyde :+1:

**Comment by kkaempf** [comment]
@Oats87 @jakefhyde - who should merge this PR ?

**Review by mjura — APPROVED** [review]
LGTM

**Review by furkatgofurov7 — DISMISSED** [review]

**Review by alexander-demicev — APPROVED** [review]

**Comment by richardcase** [comment]
I had to rebase because of a conflict.

@Oats87 @jakefhyde @alexander-demicev  -  any change of a review/approval again?

**Review by alexander-demicev — APPROVED** [review]

**Review by jakefhyde — APPROVED** [review]

**Pull request merged**